### ISSUE TITLE
Icon image ratio and zoom

### ISSUE THREAD

**Issue by VaporAPX** [Repooc/ElvUI_RatioMinimapAuras#1]
Would it be possible to add a feature whereas the icon height and width and adjustable, but the actual image in the icon keeps its aspect ratio? I really like having my icons rectangular, but in doing so it distorts the actual image of the icon. Adding a zoom effect would be great too. I'm currently showing my buffs/debuffs with Weakauras, rectangular with a 30% zoom and it looks really crisp. The problem is I can't interact with them. I'd love to be able to do that with the ElvUI buffs.

Thanks for your time!

**Comment by Repooc** [comment]
should be good for ya as this will match how the actionbars look when u disable keep aspect ratio on them

**Comment by VaporAPX** [comment]
Wow I wasn't expecting a fix that quickly, very much appreciated! It looks much better this way. Is it at all possible to add a zoom feature? or is that me being a pain in the ass and more coding than you wanna do? lol. Blizzard icons tend to leave a little black line on the sides of some icons when they are stretched to rectangular sizes.  The zoom fixes that issue, atleast in Weakauras.

![Buffs](https://user-images.githubusercontent.com/21989751/199436115-9dacaabd-b569-4a01-91a3-d925725eb10a.jpg)

Top line of buffs is weakauras, bottom is Elv running your mod. It's most noticeable on the "hand shake" icon, Sign of the Emissary, on the right side.

Apologies if I'm just being nitpicky and a pain, very much appreciate your time and efforts.

**Comment by Repooc** [comment]
meh, i dont think i will do a "zoom" effect but i will do something just as good i suppose...

https://user-images.githubusercontent.com/1479745/199461041-d674d5ee-3858-468a-a83b-556a12cdabe5.mp4

basically lets u adjust the TexCoords urself... need to add it to debuffs and clean it up and will do a release with that later on after a nap lol

**Comment by VaporAPX** [comment]
That'll be awesome! Thank you again for the work you've done with this mod, and the speedy responses to my nagging lol.  Happy I'll be able to use ElvUI buffs again, weak auras just doesn't quite cut it, even if they look pretty.

**Comment by Repooc** [comment]
Just woke up... shhh i know i slept in lol.  I am cleaning it up now and will push a release soon.

**Comment by VaporAPX** [comment]
Sleep is important! Thanks again, very much appreciated :)

**Comment by Repooc** [comment]
ok, just pushed a release and if u are using curseforge as the source, the update will be available shortly

**Comment by Repooc** [comment]
> That'll be awesome! Thank you again for the work you've done with this mod, and the speedy responses to my nagging lol. Happy I'll be able to use ElvUI buffs again, weak auras just doesn't quite cut it, even if they look pretty.

doesnt feel like nagging, id find it nagging if i said no and u kept on about it lol You caught me at the right time for the quick responses as i was already pushing updates for my other projects.  So i had everything open and it was something i could quickly do lol

**Comment by Repooc** [comment]
lol you are welcome... if you are using wowup client, u can change the source to grab from wago which will is another way you can help support the project as every download you get from wago/curse gives me just a little more points for that download 😉 (maybe grab new version once it shows up on cf app?) no worries if u dont wanna, just a bit of self promotion to help get those points lol

**Comment by VaporAPX** [comment]
I didn't realize that was a thing. I absolutely will, it's the least I can do!

**Comment by Repooc** [comment]
Yep, we get points for the downloads which we can cash out.  Those are the only 2 sites that do that.  That's why all the projects of mine are only on the 2 sites, shadow and light is the only addon that is hosted on tukui in addition to the other 2 mainly bc Darth and I did put it up there when we first added addon hosting on tukui.org and don't feel like dealing with people complaining they can't get it from there any longer if we remove it.... still thinking about removing it from there as well lol. Anyways, enjoy and thanks again.
